app/src/main/java/com/asfoundation/wallet/home/HomeViewModel.kt

```diff
@@ -0,0 +1,482 @@
+    var fiatSum = MINUS_ONE
+    if (appcoinsFiatValue > MINUS_ONE) {
+      fiatSum = appcoinsFiatValue
+    }
+    if (creditsFiatValue > MINUS_ONE) {
+      fiatSum = if (fiatSum > MINUS_ONE) {
+        fiatSum.add(creditsFiatValue)
+      } else {
+        creditsFiatValue
+      }
+    }
+    if (etherFiatValue > MINUS_ONE) {
+      fiatSum = if (fiatSum > MINUS_ONE) {
+        fiatSum.add(etherFiatValue)
+      } else {
+        etherFiatValue
+      }
+    }
+    return fiatSum
+  }
+
+  private fun updateTransactions(walletModel: TransactionsWalletModel?): Completable {
+    if (walletModel == null) return Completable.complete()
+    val retainValue = if (walletModel.isNewWallet) null else HomeState::transactionsModelAsync
+    return Completable.fromObservable(
+        Observable.combineLatest(getTransactions(walletModel.wallet), getCardNotifications(),
+            getAppcoinsApps(), getMaxBonus(), observeNetworkAndWallet(),
+            { transactions: List<Transaction>, notifications: List<CardNotification>, apps: List<AppcoinsApplication>, maxBonus: Double, transactionsWalletModel: TransactionsWalletModel ->
+              createTransactionsModel(transactions, notifications, apps, maxBonus,
+                  transactionsWalletModel)
+            })
+            .doOnNext { (transactions) ->
+              updateTransactionsNumberUseCase(
+                  transactions)
+            }
+            .subscribeOn(networkScheduler)
+            .observeOn(viewScheduler)
+            .asAsyncToState(retainValue) { copy(transactionsModelAsync = it) }
+            .map { walletModel })
+  }
+
+  private fun createTransactionsModel(transactions: List<Transaction>,
+                                      notifications: List<CardNotification>,
+                                      apps: List<AppcoinsApplication>, maxBonus: Double,
+                                      transactionsWalletModel: TransactionsWalletModel): TransactionsModel {
+    return TransactionsModel(transactions, notifications, apps, maxBonus, transactionsWalletModel)
+  }
+
+  /**
+   * Transactions are refreshed every [.UPDATE_INTERVAL] seconds, and stops while
+   * [.refreshData] is false
+   */
+  private fun getTransactions(wallet: Wallet): Observable<List<Transaction>>? {
+    return Observable.interval(0, UPDATE_INTERVAL, TimeUnit.MILLISECONDS)
+        .flatMap { observeRefreshData() }
+        .switchMap {
+          fetchTransactionsUseCase(wallet.address)
+        }
+        .subscribeOn(networkScheduler)
+        .onErrorReturnItem(emptyList())
+        .doAfterTerminate { stopFetchTransactionsUseCase() }
+  }
+
+  private fun getCardNotifications(): Observable<List<CardNotification>> {
+    return refreshCardNotifications.flatMapSingle { getCardNotificationsUseCase() }
+        .subscribeOn(networkScheduler)
+        .onErrorReturnItem(emptyList())
+  }
+
+  private fun getAppcoinsApps(): Observable<List<AppcoinsApplication>> {
```

Comment:
This method should be removed since since we are no longer showing the applications bundle in the home screen